python/tskit/trees.py

```diff
@@ -10476,6 +10476,45 @@ def ld_matrix(
             mode=mode,
         )
 
+    def individual_nodes(self):
+        """
+        Return an array of node IDs for each individual in the tree sequence.
+        Errors if the tree sequence has no individuals, if any sample doesn't
+        have an individual, if individuals have nodes that are both samples
+        and non-samples, or if an individual has no samples.
+
+        :return: Array of shape (num_individuals, max_ploidy) containing node IDs.
+            Values of -1 indicate unused slots for individuals with ploidy
+            less than the maximum.
+        :rtype: numpy.ndarray (dtype=np.int32)
+        """
+        if self.num_individuals == 0:
+            raise ValueError("Tree sequence has no individuals")
+
+        max_ploidy = 0
+        for i in range(self.num_individuals):
+            ind = self.individual(i)
```

Comment:
Just curious - why not
```suggestion
        for ind in self.individuals():
```

Reply:
okay phew thought I was missing something =)